### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by mahbubul14**
## Project requirements:

* Copy the files calculate.js and operate.js into a logic/ directory in your project.
* Analyze the files calculate.js and operate.js, be sure you understand what they do.
* Import the files in your Calculator component.
* Implement the event handlers you need to use the math logic from calculate.js and operate.js in your React component.
* Practice what you have learned until now: use state and props, pass props to child components, use lifecycle methods, lift state up.
* Test your app in the browser. It should be able to perform math operations.

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  if (operation === '÷') {
+    try {
+      return one.div(two).toString();
+    } catch (err) {
+      return "Can't divide by 0.";
+    }
+  }
+  if (operation === '%') {
+    return one.mod(two).toString();
+  }
```
Comment (arslanbisharat) [inline src/logic/operate.js:26]: - [x] Remember that the % button should work as modulus. It's not working properly.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/logic/operate.js (hunk 1 above)
Comment (mahbubul14) [reply src/logic/operate.js:26]: @arslanbisharat, I am asking for a better understanding. in the requirement, it is said "**Implement the event handlers you need to use the math logic from calculate.js and operate.js in your React component.**"
Should we need to change anything in the operate.js file? The file was not created by me.

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  if (operation === '-') {
+    return one.minus(two).toString();
+  }
+  if (operation === 'x') {
+    return one.times(two).toString();
+  }
+  if (operation === '÷') {
+    return one.div(two).toString();
+  }
```
Comment (EricMbouwe) [inline src/logic/operate.js]: - [x] The case of the division by `zero` is not handled by your function and it crashes when it happens. See the image below, let's fix that.

Screenshot from 2021-10-22 11-03-55

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
### Hi @mahbubul14 ,

Great work on making the changes requested by a previous reviewer 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

# Required Changes ♻️ ⛔️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by oloomoses — APPROVED** [review]
Hi @mahbubul14, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**